stdlib/public/core/KeyPath.swift

```diff
@@ -3703,3 +3703,138 @@ internal func _instantiateKeyPathBuffer(
   }
 }
 
+@_silgen_name("swift_keypath_dladdr")
+internal func keypath_dladdr(_: UnsafeRawPointer) -> UnsafePointer<CChar>?
+
+@_silgen_name("swift_keypathSourceString")
+internal func demangle(
+    name: UnsafePointer<CChar>
+) -> UnsafePointer<CChar>?
+
+
+fileprivate func dynamicLibraryAddress<Base, Leaf>(
+    of pointer: ComputedAccessorsPtr,
+    _: Base.Type,
+    _ leaf: Leaf.Type
+) -> String {
+    let getter: ComputedAccessorsPtr.Getter<Base, Leaf> = pointer.getter()
+    let pointer = unsafeBitCast(getter, to: UnsafeRawPointer.self)
+    if let cString = keypath_dladdr(UnsafeRawPointer(pointer)),
+        let demangled = demangle(name: cString) {
+        defer {
+            demangled.deallocate()
+        }
+        return String(cString: demangled)
+    } else {
+        return "<computed \(pointer) (\(leaf))>"
+    }
+}
+
+@available(SwiftStdLib 5.7, *)
+extension AnyKeyPath: CustomDebugStringConvertible {
+        
+    @available(SwiftStdLib 5.7, *)
+    public var debugDescription: String {
+        var description = "\\\(String(describing: Self.rootType))"
+        return withBuffer {
+            var buffer = $0
+            if buffer.data.isEmpty {
+                _internalInvariantFailure("key path has no components")
+            }
+            var valueType: Any.Type = Self.rootType
+            while true {
+                let (rawComponent, optNextType) = buffer.next()
+                let hasEnded = optNextType == nil
+                let nextType = optNextType ?? Self.valueType
+                func name(for offset: Int) -> String {
```

Comment:
I think it'd be easier to read if you removed these inline functions and instead joined the similar cases in the switch and defined these implementation in those. For example:
```swift
case .struct(let offset), .class(let offset):
  #if SWIFT_ENABLE_REFLECTION
    ...
  #endif
  ...

case .get(_, let accessors, _),
     .mutatingGetSet(_, let accessors, _),
     .nonmutatingGetSet(_, let accessors, _):
  func project<Base>(_: Base.Type) -> String {
    ...
  }

  ...
```